src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferStringContainsOverIndexOfTests.cs

```diff
@@ -0,0 +1,707 @@
+        [InlineData("a", "a", true)]
+        public async Task TestLeftOperandInvocation(string input, string fix, bool isCharTest)
+        {
+            string startQuote = isCharTest ? "'" : "\"";
+            string endQuote = isCharTest ? "'" : "\", System.StringComparison.Ordinal";
+            string csInput = @"
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            const string str = ""This is a string"";
+            if ([|str.IndexOf(" + startQuote + input + endQuote + @") == -1|])
+            {
+
+            }
+        } 
+    } 
+}";
+            var testOrdinal = new VerifyCS.Test
+            {
+                TestState = { Sources = { csInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+                LanguageVersion = CodeAnalysis.CSharp.LanguageVersion.Preview,
+            };
+            await testOrdinal.RunAsync();
+
+            startQuote = "\"";
+            string vbCharLiteral = isCharTest ? "c" : "";
+            string stringComparison = isCharTest ? "" : ", System.StringComparison.Ordinal";
+            string vbInput = @"
+Public Class StringOf
+    Class TestClass
+        Public Sub Main()
+            Dim Str As String = ""This is a statement""
+            If [|Str.IndexOf(" + startQuote + input + startQuote + vbCharLiteral + stringComparison + @") = -1|] Then
+
+            End If
+        End Sub
+    End Class
+End Class
+";
+
+            var testOrdinal_vb = new VerifyVB.Test
+            {
+                TestState = { Sources = { vbInput } },
+                ReferenceAssemblies = ReferenceAssemblies.NetCore.NetCoreApp50,
+            };
+            await testOrdinal_vb.RunAsync();
+        }
+
+        [Theory]
+        [InlineData("This", false, ", 1")]
+        [InlineData("a", true, ", 1")]
+        [InlineData("This", false, ", 1", ", 2")]
+        [InlineData("a", true, ", 1", ", 2")]
+        [InlineData("This", false, ", 1", ", System.StringComparison.OrdinalIgnoreCase")]
+        [InlineData("This", false, ", 1", ", 2", ", System.StringComparison.OrdinalIgnoreCase")]
+        public async Task TestTooManyArgumentsToIndexOf(string input, bool isCharTest, params string[] inputArguments)
+        {
+            string quotes = isCharTest ? "'" : "\"";
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < inputArguments.Length; i++)
+            {
+                sb.Append(inputArguments[i]);
+            }
+
+            string csInput = @"
+namespace TestNamespace 
+{ 
+    class TestClass 
+    { 
+        private void TestMethod() 
+        { 
+            const string str = ""This is a string"";
+            int index = str.IndexOf(" + quotes + input + quotes + sb.ToString() + @");
+            if (index == -1)
+            {
+
+            }
+        } 
+    } 
+}";
+            await VerifyCS.VerifyAnalyzerAsync(csInput);
```

Comment:
Prior tests all use `ReferenceAssemblies.NetCore.NetCoreApp50`. Why not here?

Reply:
Fixed.